dev/release/README.md

```diff
@@ -25,19 +25,24 @@ Patch releases are made on an adhoc basis, but we try and avoid them given the f
 
 ## Branching Policy
 
-- When we prepare a new release, we create a release branch, such as `branch-18` in the Apache repository (not in a fork)
+- When we prepare a new release, we create a release branch, such as `branch-37` in the Apache repository (not in a fork)
 - We update the crate version and generate the changelog in this branch and create a PR against the main branch
 - Once the PR is approved and merged, we tag the rc in the release branch, and release from the release branch
 - Bug fixes can be merged to the release branch and patch releases can be created from the release branch
 
 #### How to add changes to `branch-*` branch?
 
-If you would like to propose your change for inclusion in a release branch
+If you would like to propose your change for inclusion in a release branch for a
+patch release:
 
-1. follow normal workflow to create PR to `main` branch and wait for its approval and merges.
-2. after PR is squash merged to `main`, branch from most recent release branch (e.g. `branch-18`), cherry-pick the commit and create a PR to release branch.
+1. Find (or create) the issue for the incremental release ([example release issue]) and discuss the proposed change there with the maintainers.
```

Comment:
I tried to add some examples of what a release issue and backport PR looked like so we can follow the same example in the future